Client/Frontend/Settings/Rewards Internals/RewardsInternalsShareController.swift

```diff
@@ -0,0 +1,238 @@
+        tableView.estimatedRowHeight = UITableView.automaticDimension
+        
+        tableView.allowsMultipleSelectionDuringEditing = true
+        tableView.setEditing(true, animated: false)
+        for index in initiallySelectedSharables {
+            tableView.selectRow(at: IndexPath(row: index, section: 0), animated: false, scrollPosition: .none)
+        }
+        
+        tableView.tableFooterView = progressIndiciator
+        progressIndiciator.observedProgress = shareProgress
+        
+        progressIndiciator.sizeToFit()
+        
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(tappedCancel))
+    }
+    
+    @objc private func tappedCancel() {
+        shareProgress.cancel()
+        dismiss(animated: true)
+    }
+    
+    deinit {
+        cleanup(callingFromDeinit: true)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        progressIndiciator.frame = CGRect(x: 16, y: progressIndiciator.frame.origin.y, width: view.bounds.width - 32, height: 16)
+    }
+    
+    private let shareProgress = Progress()
+    private var shareProgressObservation: NSKeyValueObservation?
+    
+    private let progressIndiciator = UIProgressView(progressViewStyle: .default)
+    
+    private func cleanup(callingFromDeinit: Bool = false) {
+        if !callingFromDeinit {
+            DispatchQueue.main.async {
+                // Reset
+                self.isSharing = false
+                self.progressIndiciator.progress = 0
+            }
+        }
+        do {
+            if FileManager.default.fileExists(atPath: dropDirectory.path) {
+                try FileManager.default.removeItem(at: dropDirectory)
+            }
+            if FileManager.default.fileExists(atPath: zipPath.path) {
+                try FileManager.default.removeItem(at: zipPath)
+            }
+        } catch {
+            log.warning("Failed to cleanup sharing Rewards Internals files: \(error)")
+        }
+    }
+    
+    private let dropDirectory: URL
+    private let zipPath: URL
+    
+    private func share(_ senderIndexPath: IndexPath) {
+        // create temp folder, zip, share
+        guard let selectedIndexPaths = tableView.indexPathsForSelectedRows else { return }
+        let sharables = selectedIndexPaths.map { RewardsInternalsSharable.all[$0.row] }
+        
+        let dateFormatter = DateFormatter().then {
+            $0.dateStyle = .long
+        }
+        let dateAndTimeFormatter = DateFormatter().then {
+            $0.dateStyle = .long
+            $0.timeStyle = .long
+        }
+        let builder = RewardsInternalsSharableBuilder(rewards: self.rewards, dateFormatter: dateFormatter, dateAndTimeFormatter: dateAndTimeFormatter)
+        do {
+            try FileManager.default.createDirectory(at: dropDirectory, withIntermediateDirectories: true, attributes: nil)
```

Comment:
nit: `createDirectory` will be successful if the directory already exists. Deleting the directory would help ensuring that the code is not including any `stale logs`

Reply:
Good point, will add code to check if the directory already exists and delete the contents of it does

Reply:
Fixed